app/service/file_svc.py

```diff
@@ -70,6 +70,18 @@ async def create_exfil_sub_directory(self, dir_name):
             os.makedirs(path)
         return path
 
+    async def create_exfil_operation_directory(self, dir_name, agent_name):
+        op_list = self.data_svc.ram['operations']
+        op_list_filtered = [x for x in op_list if x.state not in x.get_finished_states()]
+        timestamp_chars = str.maketrans({'T': '_', ':': ''})
+        special_chars = {ord(c): '_' for c in r':<>"/\|?*'}
+        agent_opid = [(x.name.translate(special_chars), '_', x.start.strftime("%Y-%m-%dT%H:%M:%SZ").translate(timestamp_chars))
```

Comment:
use a utility function for this time conversion, there should be one that exists already

Reply:
The issue was the colons in the timestamp not being OS safe. Couldn't find a time utility function to remove the colons. However, I simplified the code and removed the translation with a direct conversion `"%Y-%m-%d_%H%M%SZ"` instead of `"%Y-%m-%dT%H:%M:%SZ").translate(timestamp_chars)`